### PR TITLE
docker: add vdsm user to container image

### PR DESCRIPTION
**Pull request by aesteve-rh**
Create an unprivileged vdsm user in the container image to enable running tests as non-root.

This could be achieved by using other unprivileged user, e.g., `nobody`, but creating vdsm allows creating tests specific for vdsm user that run both in the pipelines and in the test VMs.

Signed-off-by: Albert Esteve <aesteve@redhat.com>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,11 +5,27 @@ env:
 jobs:
   lint:
     runs-on: ubuntu-latest
-    container: quay.io/ovirt/vdsm-test-centos-8
+    container: quay.io/alesgar/vdsm-test-centos-8
```
Comment (nirs) [inline .github/workflows/ci.yml:8]: This is temporary change for testing I guess?

File: docker/Dockerfile.centos-8 [hunk 2]
```diff
@@ -2,7 +2,7 @@ FROM quay.io/centos/centos:stream8
 
 # Add runtime dependencies.
 RUN dnf install -y \
-        http://resources.ovirt.org/pub/yum-repo/ovirt-release-master.rpm \
+        http://resources.ovirt.org/pub/yum-repo/ovirt-release44.rpm \
```
Comment (nirs) [inline docker/Dockerfile.centos-8:5]: This does not look right.

File: docker/Dockerfile.centos-8 [hunk 3]
```diff
@@ -60,6 +60,9 @@ RUN dnf install -y \
         xfsprogs \
     && dnf clean all
 
+# Create vdsm user with groups kvm,qemu,sanlock
+RUN useradd --system -m -u 36 -N -G 36,107,179 vdsm
```
Comment (nirs) [inline docker/Dockerfile.centos-8:64]: This like the only change needed in this pr, added in all dockerfiles.

File: docker/Makefile [hunk 4]
```diff
@@ -30,5 +30,5 @@ $(targets):
 
 push:
 	for name in $(targets); do \
-		podman push $(prefix)-$$name ovirtorg/$(prefix)-$$name; \
+		podman push $(prefix)-$$name ovirt/$(prefix)-$$name; \
```
Comment (nirs) [inline docker/Makefile:33]: This push url is missing quay.io/ prefix, and should really be in its own commit.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -5,11 +5,27 @@ env:
...
+      # Required to create loop devices.
+      options: --privileged
+    steps:
+    - uses: actions/checkout@v2
+    - name: Run storage tests
+      run: |
+        ./automation/setup-storage.sh
+        chown -R vdsm:kvm ./ /var/tmp/vdsm*
+        su -c "tox -e storage -- --cov-fail-under 60 \
+           -m 'not (integration or slow or stress or root)' storage" -- vdsm
```
Comment (nirs) [inline .github/workflows/ci.yml:28]: I think we need:
- tests-storage
- tetsts-storage-root

Both can use the same tests-storage.sh script, with additional argument so the script can run tests as root or as regular user.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (aesteve-rh) [reply .github/workflows/ci.yml:8]: Right. All changes in the ci.yml and the setup-storage script will go away. I have another branch for the test split, this branch is only for the image update. Mentioned it in the PR text.

File: docker/Dockerfile.centos-8 (hunk 2 above)
Comment (aesteve-rh) [reply docker/Dockerfile.centos-8:5]: release-master was not available, and this was the most recent rpm I saw in the same resources folder. Maybe we need to upload new rpms? Not sure.

File: docker/Dockerfile.centos-8 (hunk 3 above)
Comment (aesteve-rh) [reply docker/Dockerfile.centos-8:64]: 👍

File: .github/workflows/ci.yml (hunk 5 above)
Comment (aesteve-rh) [reply .github/workflows/ci.yml:28]: Was a test for running with vdsm user, I will remove the change before merging. I have it covered in a different branch and is as you suggest, two jobs for storage testing.

File: docker/Makefile (hunk 4 above)
Comment (aesteve-rh) [reply docker/Makefile:33]: Right, I’ll separate it.

File: docker/Dockerfile.centos-8 (hunk 2 above)
Comment (nirs) [reply docker/Dockerfile.centos-8:5]: The issue is old links in the docker files. The right way to install is explained
in ovirt master snapshot copr repo. You need to update all the docker files to consume
the right repos.
https://copr.fedorainfracloud.org/coprs/ovirt/ovirt-master-snapshot/

Comment (aesteve-rh) [reply docker/Dockerfile.centos-8:5]: Ah I see, I should've known.

I'll update them :+1:

**Review by nirs — CHANGES_REQUESTED** [review]

File: docker/Dockerfile.centos-8 [hunk 6]
```diff
@@ -49,7 +49,7 @@ RUN dnf install -y \
         python3-sanlock \
         python3-six \
         python3-yaml \
-        qemu-img-6.0.0 \
+        qemu-img-6.1.0 \
```
[the remark this answers is not in the document]
Comment (aesteve-rh) [reply docker/Dockerfile.centos-8]: Done, and tested the new image in the pipelines :)

**Review by nirs — CHANGES_REQUESTED** [review]

**Comment by aesteve-rh** [comment]
Done, just pushed it. I'll trigger a run with a follow-up branch I have with updates in the CI :)
